cmdstanpy/utils.py

```diff
@@ -1431,3 +1432,29 @@ def __enter__(self) -> Tuple[str, bool]:
     def __exit__(self, exc_type, exc_val, exc_tb) -> None:  # type: ignore
         if self._tmpdir:
             shutil.rmtree(self._tmpdir, ignore_errors=True)
+
+
+def write_stan_file(code: str, directory: Union[str, None] = None) -> str:
+    """
+    Write the model code to a file and return the path based on the code hash.
+
+    :param code: Code to write to a file.
+    :param directory: Directory to write the code to (defaults to the cmdstanpy
+        models directory).
+
+    :return: Path to the file containing the code..
+    """
+    # Generate the path based on the code and check whether it already exists.
+    directory = os.path.abspath(
+        directory or os.path.join(tempfile.gettempdir(), 'cmdstanpy_models')
```

Comment:
Cmdstanpy makes its own tempdir we use for everything - https://github.com/stan-dev/cmdstanpy/blob/683a135088eb48a8372fea9837ef96805aeaca10/cmdstanpy/__init__.py#L21
we should use it here too

Reply:
Maybe slightly premature commit on my part. We can either prefix the filename (e.g. to get filenames like `model_{hash}.stan`) or have a subdirectory (as in the current commit). Any preferences?

Reply:
I think I prefer `model_`